modules/conflict_resolver/jsx/react.conflict_resolver.js

```diff
@@ -0,0 +1,259 @@
+ * 	module.
+ */
+
+/**
+ * @doc function
+ * @name updateState
+ * @param {object} filter to update the state.
+ * @description This function is responsible for updating
+ * the UnresolvedConflictsPane state from the parent.
+ */
+var updateState = function(filter) {
+  this.setState({filter});
+};
+
+/**
+ *	This is the React class for Unresolved Conflicts
+ */
+class UnresolvedConflictsPane extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      Data: this.props.data
+    };
+    updateState = updateState.bind(this);
+  }
+
+  // Render the HTML
+  render() {
+    return (
+      <TabPane Title="" {...this.props}>
+        <form method="post" action={this.props.url.base + "/conflict_resolver/"} name="conflict_resolver" id="conflict_resolver">
+
+          <RDynamicDataTable
+            DataURL={this.props.url.data.unresolved}
+            Data={this.state.Data.Data}
+            Headers={this.state.Data.Headers}
+            Filter={this.state.filter}
+            getFormattedCell={formatColumn}
+          />
+
+          <div className="pull-right">
+            <input className="btn btn-sm btn-primary" name="fire_away" value="Save" type="submit"/>
+            <input className="btn btn-sm btn-primary" value="Reset" type="reset" style={{marginLeft: 3 + 'px'}}/>
+          </div>
+
+        </form>
+      </TabPane>
+    );
+  }
+}
+UnresolvedConflictsPane.propTypes = {
+  url: React.PropTypes.object.isRequired,
+  data: React.PropTypes.object.isRequired
+};
+UnresolvedConflictsPane.defaultProps = {
+  module: '',
+  url: {
+    base: '',
+    data: {
+      unresolved: '',
+      resolved: ''
+    }
+  },
+  data: {}
+};
+
+/**
+ * This is the React class for the conflict resolver
+ */
+class ConflictResolverApp extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    loris.hiddenHeaders = [
+      'Value1',
+      'Value2',
+      'Hash',
+      'Site'
+    ];
+
+    this.state = {
+      isLoaded: false,
+      filter: {},
+      form: {}
+    };
+
+    // Bind component instance to custom methods
+    this.fetchData = this.fetchData.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   */
+  fetchData() {
+    $.ajax(this.props.url.data.unresolved, {
+      method: 'GET',
+      dataType: 'json',
+      success: function(data) {
+        this.setState({
+          Data: data,
+          isLoaded: true
+        });
+      }.bind(this),
+      error: function(error) {
+        console.error(error);
+      }
+    });
+  }
+
+  updateFilter(filter) {
+    this.setState({filter});
+    updateState(filter);
```

Comment:
Is this not redundant since what `updateState(filter)` does is update the state of the filter in the same manner that line `128` does?

Reply:
@HenriRabalais The function "updateState" is being used to pass the filtered state between the two ConflictResolverApp and the UnresolvedConflictsPane components.

Reply:
Hmm, I would normally say that this should be done through a call-back function from the parent component. But it seems in this case you don't have a parent `index` component. Ideally, both Filters should have a shared parent component, but perhaps that can be next reactification step when we lift the state of every module.

@davidblader @xlecours @ZainVirani @driusan  thoughts?

Reply:
Shouldn't the filter be passed as a prop to `UnresolvedConflictsPane` after linue `158`?

Reply:
i agree with henri. we should be passing states through props

Reply:
When I made that comment three weeks ago, I tested the PR and managed to get the filters to work by passing the filter as a `prop`. I suggest going that route since it's best for us to be passing `props` down from a top level stateful controller.

@davidblader @driusan @davidblader perhaps you have some input on this?

Reply:
not sure i understand @intralizee 's explanation

in the browser for a given tab, i see two components, `ConflictResolverApp` and either `UnresolvedConflictsPane` or `ResolvedConflictsPane`. updating the filter causes a `filter` object to be duplicated in the states of both `ConflictResolverApp` and `UnresolvedConflictsPane` / `ResolvedConflictsPane`

duplicating state like this is generally a bad idea, i would recommend following @HenriRabalais 's feedback